src/main/scala/files_metadata/application/FilesMetaUseCases.scala

```diff
@@ -228,4 +228,31 @@ class FilesMetaUseCases {
 
     repository.updateFileParent( fileUUID, newParentUUID )
   }
+
+  def unShareFile(
+      ownerUUID: UUID,
+      fileUUID: UUID,
+      otherUserUUID: UUID
+  ): Unit = {
+    val fileMeta = repository.getFileMeta( fileUUID )
+
+    if (fileMeta.ownerUuid != ownerUUID) {
+      throw DomainExceptions.FileNotOwnedException(
+        "The user does not own the file"
+      )
+    }
+    if (ownerUUID == otherUserUUID) {
+      throw DomainExceptions.FileNotOwnedException(
+        "The user and the other user are the same"
```

Comment:
Try to write errors than can be shown in the UI. Instead of "The user and the other user are the same" write "You cannot un-share a file with yourself".

Reply:
Repaired:
![image](https://github.com/hawks-atlanta/metadata-scala/assets/47455237/505ea475-1499-4cd1-87e5-43cb690bfc43)